File: BAEKJOON/data_structure/javascript/1874.js
const fs = require("fs");
let input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().trim()
    : fs
        .readFileSync(__dirname + "/input.txt")
        .toString()
        .trim();

input = input.split("\n").map((value) => Number(value));
const n = input.shift();

function solution(input) {
  const stack = [];
  const answer = [];
  let current = 1;
  let cursor;

  for (let number of input) {
    cursor = current;

    for (let i = cursor; i <= number; i++) {
      stack.push(i);
      answer.push("+");
      current++;
    }

    const popedNumber = stack.pop();
    if (popedNumber !== number) {
      return "NO";
    } else {
      answer.push("-");
    }
  }

  return answer.join("\n");
}

console.log(solution(input));

File: BAEKJOON/data_structure/python/1874.py
""" 
22.07.06
BOJ : 1874 / 자료구조
"""
import sys

n = int(input())
stack = []
current_num = 1
result = []

for _ in range(n):
    num = int(sys.stdin.readline())

    #숫자가 같을 때 까지 push
    while current_num <= num:
        stack.append(current_num)
        current_num += 1
        result.append('+')
    
    # stack의 top이 num과 같으면 pop
    if stack[-1] == num:
        stack.pop()
        result.append('-')

if stack == []:
    print('\n'.join(result))
else:
    print('NO')
